File: app/schemas/onboarding.py
"""Onboarding API schemas (Phase AI.86)."""

from __future__ import annotations

from pydantic import BaseModel, Field

from app.schemas.contracts import OnboardingStep


class OnboardingStepStatus(BaseModel):
    step: OnboardingStep
    completed: bool
    derived: bool = True
    manual_allowed: bool = False


class OnboardingStatusResponse(BaseModel):
    project_id: str | None = None
    steps: list[OnboardingStepStatus] = Field(default_factory=list)
    completed_count: int = 0
    total_count: int = 0


class OnboardingCompleteStepRequest(BaseModel):
    step: OnboardingStep

File: web/src/lib/api/types/onboarding.ts
export type OnboardingStep =
  | "project_created"
  | "agents_seeded"
  | "demo_seeded"
  | "first_chat_done"
  | "first_asset_created"
  | "first_publication_job_created";

export type OnboardingStepStatus = {
  step: OnboardingStep;
  completed: boolean;
  derived: boolean;
  manual_allowed: boolean;
};

export type OnboardingStatus = {
  project_id: string | null;
  steps: OnboardingStepStatus[];
  completed_count: number;
  total_count: number;
};
